producer.go

```diff
@@ -106,48 +107,73 @@ type AsyncApexLogNSQHandler struct {
 // be published to.
 //
 func NewAsyncApexLogNSQHandler(marshalFunc MarshalFunc, publishFunc PublishFunc, topic string, bufferSize int) *AsyncApexLogNSQHandler {
+	var backoff = 0
 	logChan := make(chan *log.Entry, bufferSize)
 	stopChan := make(chan bool, 1)
+
+	incrementBackoff := func(backoff int) int {
+		if backoff == 0 {
+			return 1
+		}
+		newBackoff := backoff * 2
+		if newBackoff > maximumBackoffMultiple {
+			return maximumBackoffMultiple
+		}
+		return newBackoff
+	}
+
+	handler := &AsyncApexLogNSQHandler{
+		logChan:  logChan,
+		stopChan: stopChan,
+	}
+
 	go func(cLog chan *log.Entry, cStop chan bool) {
 		var e *log.Entry
 		for {
```

Comment:
by backing off in the for loop it means we discard log lines for temporariy failures. if instead you used the backoff pattern we've used elsewhere you can ensure temporary failures don't result in dropped messages.

https://play.golang.org/p/oKCavXlE4VH